src/economy/shipping_schedule.cc

```diff
@@ -0,0 +1,1729 @@
+	}
+}
+
+// Shared logic for steps 5.1 and 5.3
+void ShippingSchedule::load_on_ship(Game& game,
+                                    PrioritisedPortPair& ppp,
+                                    std::list<PrioritisedPortPair>& all_ppps) {
+	const uint32_t take = std::min(ppp.open_count, ppp.ships.front().capacity);
+	Ship* ship = ppp.ships.front().ship;
+	sslog("load_on_ship: PPP %u –> %u (open_count %u): assigning %u items (capacity %u) to %s\n",
+	      ppp.start->serial(), ppp.end->serial(), ppp.open_count, take, ppp.ships.front().capacity,
+	      ship->get_shipname().c_str());
+	assert(take);
+	// We assume that EITHER both end points are already part of the plan,
+	// or that the start point is the last entry in the plan
+	if (plans_[ship].back().dock == ppp.start) {
+		assert(!plans_[ship].back().expedition);
+		auto it = plans_[ship].back().load_there.find(ppp.end);
+		if (it != plans_[ship].back().load_there.end()) {
+			it->second += take;
+		} else {
+			plans_[ship].back().load_there[ppp.end] = take;
+		}
+		Path path;
+		int32_t d = -1;
+		fleet_.get_path(*ppp.start, *ppp.end, path);
+		game.map().calc_cost(path, &d, nullptr);
+		assert(d >= 0);
+		plans_[ship].push_back(SchedulingState(ppp.end, false, d));
+	} else {
+		for (SchedulingState& ss : plans_[ship]) {
+			if (ss.dock == ppp.start) {
+				auto it = ss.load_there.find(ppp.end);
+				if (it != ss.load_there.end()) {
+					it->second += take;
+				} else {
+					ss.load_there[ppp.end] = take;
+				}
+				break;
+			}
+		}
+	}
+	ppp.ships.front().capacity -= take;
+	ppp.open_count -= take;
+	if (!ppp.ships.front().capacity) {
+		ppp.ships.erase(ppp.ships.begin());
+	}
+	// update all other PPPs that were considering to load wares onto this ship
+	for (PrioritisedPortPair& p : all_ppps) {
+		auto ship_it = p.ships.end();
+		for (auto it = p.ships.begin(); it != p.ships.end(); ++it) {
+			if (it->ship == ship) {
+				ship_it = it;
+				break;
+			}
+		}
+		if (ship_it == p.ships.end()) {
+			continue;
+		}
+		// recalc capacity and score for this ship and reorder list
+		bool found_start = false;
+		bool found_end = false;
+		bool expedition = false;
+		bool start_is_last = false;
+		Duration arrival_time = 0;
+		Duration detour_start_end = 0;
+		uint32_t free_capacity = 0;
+		get_free_capacity_between(game, *ship, plans_.at(ship), *p.start, *p.end, found_start,
+		                          found_end, expedition, start_is_last, arrival_time,
+		                          detour_start_end, free_capacity);
+		p.ships.erase(ship_it);
+		if (free_capacity && found_start && !expedition && (found_end || start_is_last)) {
+			ScoredShip updated_ship(ship, free_capacity, arrival_time, detour_start_end);
+			sslog("load_on_ship: PPP %u –> %u (open_count %u) UPDATED: may assign up to %u items "
+			      "to %s (score %lu)\n",
+			      p.start->serial(), p.end->serial(), ppp.open_count, free_capacity,
+			      ship->get_shipname().c_str(), updated_ship.score);
```

Comment:
Gives me thhis error.
`../src/economy/shipping_schedule.cc:651:40: warning: format specifies type 'unsigned long' but the argument has type 'uint64_t' (aka 'unsigned long long') [-Wformat]
                              ship->get_shipname().c_str(), updated_ship.score);`